lib/ts/recipe/thirdparty/utils.ts

```diff
@@ -43,7 +35,40 @@ export function validateAndNormaliseUserInput(
         ...config.override,
     };
 
+    let emailService = config?.emailDelivery?.service;
+    /**
+     * following code is for backward compatibility.
+     * if user has not passed emailDelivery config, we
+     * use the createAndSendCustomEmail config. If the user
+     * has not passed even that config, we use the default
+     * createAndSendCustomEmail implementation
+     */
+    if (emailService === undefined) {
+        emailService = new BackwardCompatibilityService(
+            recipeInstance.recipeInterfaceImpl,
```

Comment:
By the time this comes here, `recipeInstance.recipeInterfaceImpl` is `undefined`. So this will not work.

Ways to solve this:
- pass the `recipeInstance` to the `BackwardCompatibilityService` (like we used to before)
- In normalised config, you keep `emailDelivery` as optional, and if it is, then in the recipe.ts of this recipe, you set it's emailDelivery instance's service to create a `new BackwardCompatibilityService` in there.
- Change the type of normalisedInput for emailDeliver to not be an object, but be a function that returns an emailDelivery like `getEmailDelivery: () => EmailDelivery<..>`. So in here, you make it:
   ```
   emailyDelivery = () => { return {...config?.emailDelivery,  service: new BackwardCompatibilityService(...)} }
   ```
   Then the constructor in recipe.ts will call `config.getEmailDelivery()` and at that point in time, `recipeInstance.recipeInterfaceImpl`, `recipeInstance.isInServerlessEnv` will have been initialised.


At the moment, I prefer option (3) the most. Also, can you think of a way to structure this so that this type of issue doesn't happen at all? -> For example, `isInServerlessEnv` is only used when actually sending the email. So instead of passing it in the constructor of this, we pass it as an argument to the `sendEmail` function (by putting it in `TypeEmailVerificationEmailDeliveryInput`)